shm/Ship.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <string>
+
+#include "Cargo.hpp"
+
+class Ship {
+public:
+    Ship()
+        : capacity_()
+        , maxCrew_()
+        , crew_()
+        , speed_()
+        , name_()
+        , cargo_()
+        , id_(-1)
+    {}
+
+    Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id)
+        : capacity_(capacity)
+        , maxCrew_(maxCrew)
+        , crew_(0)
+        , speed_(speed)
+        , name_(name)
+        , cargo_()
+        , id_(id)
+    {}
+    
+    Ship(int maxCrew, int speed, size_t id)
+        : Ship(0, maxCrew, speed, "", id)
+    {}
```

Comment:
I think you can reorganize C'tors. Only one should initialize fields. And rest of them should call it with some default values.